angular/projects/spark-angular/src/lib/components/sprk-masthead/sprk-masthead-selector/sprk-masthead-selector.component.ts

```diff
@@ -0,0 +1,414 @@
+          >
+            <div *ngIf="choice.content; then content; else link"></div>
+            <ng-template #link>
+              <a
+                *ngIf="!choice.routerLink"
+                sprkLink
+                variant="unstyled"
+                [attr.href]="choice.href"
+                [analyticsString]="choice.analyticsString"
+                [ngClass]="{
+                  'sprk-c-Dropdown__link': true,
+                  'sprk-c-Dropdown__link--active': choice.active
+                }"
+                [attr.aria-label]="choice.text"
+                >{{ choice.text }}
+              </a>
+              <a
+                *ngIf="choice.routerLink"
+                sprkLink
+                variant="unstyled"
+                [routerLink]="choice.routerLink"
+                [analyticsString]="choice.analyticsString"
+                [ngClass]="{
+                  'sprk-c-Dropdown__link': true,
+                  'sprk-c-Dropdown__link--active': choice.active
+                }"
+                [attr.aria-label]="choice.text"
+                >{{ choice.text }}
+              </a>
+            </ng-template>
+            <ng-template #content>
+              <a
+                sprkLink
+                *ngIf="!choice.routerLink"
+                variant="unstyled"
+                [attr.href]="choice.href"
+                [analyticsString]="choice.analyticsString"
+                [ngClass]="{
+                  'sprk-c-Dropdown__link': true,
+                  'sprk-c-Dropdown__link--active': choice.active
+                }"
+                [attr.aria-label]="choice.content.title"
+              >
+                <p class="sprk-b-TypeBodyOne">{{ choice.content.title }}</p>
+                <p sprkText variant="bodyTwo" *ngIf="choice.content.infoLine1">
+                  {{ choice.content.infoLine1 }}
+                </p>
+                <p sprkText variant="bodyTwo" *ngIf="choice.content.infoLine2">
+                  {{ choice.content.infoLine2 }}
+                </p>
+              </a>
+              <a
+                sprkLink
+                *ngIf="choice.routerLink"
+                variant="unstyled"
+                [routerLink]="choice.routerLink"
+                [analyticsString]="choice.analyticsString"
+                [ngClass]="{
+                  'sprk-c-Dropdown__link': true,
+                  'sprk-c-Dropdown__link--active': choice.active
+                }"
+                [attr.aria-label]="choice.content.title"
+              >
+                <p class="sprk-b-TypeBodyOne">{{ choice.content.title }}</p>
+                <p>{{ choice.content.infoLine1 }}</p>
+                <p>{{ choice.content.infoLine2 }}</p>
+              </a>
+            </ng-template>
+          </li>
+        </ul>
+        <ng-content select="[sprkMastheadSelectorFooter]"></ng-content>
+      </div>
+    </div>
+  `,
+})
+export class SprkMastheadSelectorComponent implements OnChanges {
```

Comment:
Let's think about what this component actually needs from the inputs, since it's internal. We don't use a lot of these and it overcomplicates the component. If there is a good use case to have them in case we need them in the future, we can make that decision, but a lot of these can be removed.